vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -75,8 +75,14 @@ def cloud_ems_hosts_attributes
         return hosts_attributes unless clouds
 
         clouds.each do |cloud_ems|
-          connection = cloud_ems.connect
-          compute_hosts  = connection.hosts.select { |x| x.service_name == "compute" }
+          connection, compute_hosts = nil
+          next unless cloud_ems_safe_request("cloud ems connection failed for [#{cloud_ems.inspect}]") do
```

Comment:
Actually it should be definitely StartdardError, not exception. For now it's safer to catch them all. QE already saw many failures (timeout, exceeded memory, ..) and we don't want anything on that side to stop the Refresh. 

I change it to what you propose, the exception should still have all the details of the failure I guess. I tried to do clear log of what went wrong exactly.